slyd/slyd/resources/extractors.py

```diff
@@ -0,0 +1,79 @@
+from slybot.fieldtypes import FieldTypeManager
+from slybot.validation.schema import get_schema_validator
+
+from .models import ExtractorSchema
+from .utils import _read_extractors, gen_id
+from ..errors import BadRequest
+from ..utils.projects import ctx, init_project
+EXTRACTOR_TYPES = set(FieldTypeManager._TYPEMAP.keys())
+
+
+@init_project
+def list_extractors(manager, attributes=None):
+    extractors = [e for id, e in _read_extractors(manager).items()]
+    context = ctx(manager)
+    return ExtractorSchema(many=True, context=context).dump(extractors).data
+
+
+@init_project
+def get_extractor(manager, extractor_id, attributes=None):
+    extractors = _read_extractors(manager)
+    extractor = extractors[extractor_id]
+    return ExtractorSchema(context=ctx(manager)).dump(extractor).data
+
+
+def create_extractor(manager, attributes):
+    extractors = _read_extractors(manager)
+    _id = gen_id(disallow=set(extractors.keys()))
+    attrs = attributes.get('data', {}).get('attributes', {})
```

Comment:
attrs is loaded from attributes. Even with setdefault it would be necessary to have a second setdefault to be sure it wouldn't throw a key error.
We could do this instead too:
```python
try:
    attrs = attributes['data']['attributes']
except KeyError:
    attrs = {}